src/OrchardCore.Modules/OrchardCore.Users/GraphQL/Startup.cs

```diff
@@ -10,6 +10,6 @@ public class Startup : StartupBase
     public override void ConfigureServices(IServiceCollection services)
     {
         services.AddSingleton<ISchemaBuilder, CurrentUserQuery>();
-        services.AddTransient<UserType>();
+        services.AddScoped<UserType>();
```

Comment:
Are you sure you get the right instance of the `UserType` in any circumstances? I think we should change the way the `UserType` is build, what I did within the `CurrentUserQuery` seems not enough anymore.

Reply:
Yes , Yes, they have the same hash code
![image](https://github.com/OrchardCMS/OrchardCore/assets/15613121/5040ae85-0998-4b46-9593-fd16f7dd8176)